### PR TITLE
Remove global variables from kubletclient

### LINKED ISSUE #673: kubelet client refactoring: optimization and global variable
@aojea has recommended a few fixes that look to be solid improvements...

1. We're using [these global variables](https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/pkg/kubeletclient/kubeletclient.go#L23-L26) which is a bad practice, additionally they're being mutated in the kubeletclient_test.go package, and there are some tests which may be running in such a way as to flake.

2. The https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/pkg/kubeletclient/kubeletclient.go#L44 `util` package is really only being used for `util.LocalEndpoint()` which is really just a couple methods that manipulate strings, found here @ https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cri/remote/util/util_unix.go#L82-L128 -- we could omit having to include the whole package and instead include these functions.

### PR DESCRIPTION
**Pull request by s1061123**
Fixes #673

### CODE REVIEW

**Review by dougbtv — APPROVED** [review]
LGTM! Thanks for straightening this up.

File: pkg/kubeletclient/kubeletclient.go [hunk 1]
```diff
@@ -18,27 +17,26 @@ import (
...
 )
 
 // GetResourceClient returns an instance of ResourceClient interface initialized with Pod resource information
-func GetResourceClient() (types.ResourceClient, error) {
+func GetResourceClient(kubeletSocket string) (types.ResourceClient, error) {
 	// If Kubelet resource API endpoint exist use that by default
 	// Or else fallback with checkpoint file
-	if hasKubeletAPIEndpoint() {
+	if kubeletSocket == "" {
+		kubeletSocket = util.LocalEndpoint(defaultPodResourcesPath, podresources.Socket)
```
Comment (aojea) [inline pkg/kubeletclient/kubeletclient.go:28]: this returns a string with the scheme of the endpoint i.e
"unix:/pkg/kubelet.sock"
https://play.golang.org/p/zqf-bAWCnVt
does os.Stat() inside `hasKubeletAPIEndpoint()` detects it?

Comment (s1061123) [reply pkg/kubeletclient/kubeletclient.go:28]: oh, good catch! Thank you for your comment! will fix it.

**Comment by aojea** [comment]
lgtm

**Review by e0ne — APPROVED** [review]

**Pull request merged**